### DISCUSSION TITLE
Negative query boosting supported for DocumentQueries?

### DISCUSSION THREAD

**Discussion by kamranayub** [ravendb/ravendb#18694] [Q&A] [answered]
I have a situation where I'd like to de-boost results if they match a condition using a negative boost. This is working in RQL, but if I try to use a `DocumentQuery` and provide a negative value to `.Boost(boost)` I get an exception:

> Boost factor must be a non-negative number
> Parameter name: boost

Is this officially supported or is there a way to somehow pass the raw boost value to RQL?

Example code:

```
searchQuery = searchQuery.OrElse()
    .OpenSubclause()
        .WhereEquals(nameof(Games_ByMetadata.Result.ExactSlugs), criteria.Slug)
        .AndAlso()
        .WhereLessThanOrEqual(x => x.Popularity, 0)
    .CloseSubclause()
    .Boost(-100);
```

NOTE: Negative boost also works in an index 😄

**Comment by ayende** [comment]
```
boosting factor where 1.0 is default, less than 1.0 is lower weight, greater than 1.0 is higher weight
```

In other words, don't do negative, Do a fraction.

```
searchQuery = searchQuery.OrElse()
    .OpenSubclause()
        .WhereEquals(nameof(Games_ByMetadata.Result.ExactSlugs), criteria.Slug)
        .AndAlso()
        .WhereLessThanOrEqual(x => x.Popularity, 0)
    .CloseSubclause()
    .Boost(0.1);
```

**Reply by kamranayub** [reply]
Yes, this works but in my case that wasn't enough, probably due to the magnitude of boosts in other places in the query. The negative boost was the only way I got what I wanted.

However, I think if I adjusted all my boost values, that would work.

Rather than doing that, I just made a more specific subclause with a higher boost value that did the trick. 

If the index _supports_ negative values though, shouldn't the client allow it as well? Is there a side effect to allowing negative boost values?

**Reply by ayende** [reply, marked as answer]
That particular check was added 14 years ago, here:

https://github.com/ravendb/ravendb/commit/fc70189675fb87518f9ee4ee6ae94c2d0748d8d3

I think you may be able to do that by setting boost to zero.

https://github.com/ravendb/ravendb/pull/8794

That may give you the effect you want.

See here: https://www.mail-archive.com/lucene-user@jakarta.apache.org/msg08868.html

Negative boost is not well defined for Lucene, and was flat out forbidden in Lucene 8.0

**Reply by kamranayub** [reply]
Ah that makes sense then. Perhaps Corax could support it, if the Lucene functionality should be preserved. 

I noticed [ElasticSearch allows negative boosting](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-boosting-query.html), so it could be a nice feature for Corax to support since that is no longer tied to Lucene semantics.
